src/main/java/gov/nasa/pds/tools/label/LabelValidator.java

```diff
@@ -312,6 +333,26 @@ public synchronized void validate(ProblemHandler handler, URL url)
 
   }
 
+  private Boolean determineSchematronValidationFlag(URL url) {
+      // Function return true if the validation against the schematron should be done or not.
+      // Note: schematron validation can be time consuming.  Only the bundle or collection should be validated against schematron.
+      // If the flag skipProductValidation is true, the labels belong to data files should not be done.
+      String nameOnly = Paths.get(url.getPath()).getFileName().toString();
+      Boolean validateAgainstSchematronFlag;
+      if (!this.skipProductValidation) {
+          validateAgainstSchematronFlag = true;
+      } else {
+          // If skip product validation is true, perform schematron validation bundle/collection labels only.
+          if (nameOnly.contains("bundle") || nameOnly.contains("collection")) {
```

Comment:
@qchaupds it is not required that a bundle or collection contain that information in the file name. if we want to check that, we need to actually read the XML and determine if it it's root is `Product_Bundle` or `Product_Collection`. depending upon the objects available in this class, you may be able to check what type of object this is? not sure what the easiest way to do this is.

Reply:
Working on the fix to inspect the file itself to see what type it is.  Will be ready for next commit.